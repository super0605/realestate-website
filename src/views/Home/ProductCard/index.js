import React from "react";
import PropTypes from "prop-types";
import NumberFormat from "react-number-format";
import {
  Container,
  ProductImg,
  Descriptoin,
  ValueGroup,
  ValueElm,
  DividerH,
  Overlay,
  OverlayContent,
  Price,
  PricePerSqm,
} from "./styled";

const ProductCard = ({
  title,
  price,
  pricePerSqm,
  sqm,
  numberOfBedrooms,
  numberOfBathrooms,
  imgUrl,
}) => {
  return (
    <Container>
      <ProductImg>
        {imgUrl && <img alt="example" src={imgUrl} />}
        <Overlay>
          <OverlayContent>
            {price && (
              <Price>
                <NumberFormat
                  displayType={"text"}
                  value={price}
                  thousandSeparator={true}
                  decimalScale={0}
                  suffix="€"
                />
              </Price>
            )}
            {pricePerSqm && (
              <PricePerSqm>
                <NumberFormat
                  displayType={"text"}
                  value={pricePerSqm}
                  thousandSeparator={true}
                  decimalScale={3}
                  suffix="€/m²"
                />
              </PricePerSqm>
            )}
          </OverlayContent>
        </Overlay>
      </ProductImg>
      {title && <Descriptoin>{title}</Descriptoin>}
      <ValueGroup>
        {sqm && (
          <ValueElm>
            <i className="fa fa-check-circle" /> {sqm} m²
          </ValueElm>
        )}
        <DividerH />
        {numberOfBedrooms ? (
          <ValueElm>
            <i className="fa fa-bed" /> {numberOfBedrooms} habs.
          </ValueElm>
        ) : (
          <ValueElm>N/A</ValueElm>
        )}
        <DividerH />
        {numberOfBathrooms ? (
          <ValueElm>
            <i className="fa fa-bath" /> {numberOfBathrooms} banos
          </ValueElm>
        ) : (
          <ValueElm>N/A</ValueElm>
        )}
      </ValueGroup>
    </Container>
  );
};

ProductCard.propTypes = {
  title: PropTypes.string,
  price: PropTypes.number,
  pricePerSqm: PropTypes.number,
  sqm: PropTypes.number,
  numberOfBedrooms: PropTypes.number,
  numberOfBathrooms: PropTypes.number,
  imgUrl: PropTypes.string,
};

export default ProductCard;
