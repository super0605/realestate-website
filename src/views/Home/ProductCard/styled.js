import styled from "styled-components";
import Divider from "../../../components/Divider";

export const Container = styled.div`
  background-color: #ffffff;
  min-height: 143px;
  margin-bottom: 24px;
  cursor: pointer;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.06), 0 1px 0 0 rgba(0, 0, 0, 0.07),
    0 2px 1px -1px rgba(0, 0, 0, 0.06);

  &:hover {
    box-shadow: rgba(0, 0, 0, 0.15) 0px 0px 10px, rgba(0, 0, 0, 0.15) 10px 10px 10px;
  }
`;

export const ProductImg = styled.div`
  width: 100%;
  height: 200px;
  background-color: #ffffff;
  min-height: 150px;
  position: relative;
  & img {
    width: 100%;
    height: 100%;
    object-fit: cover;
  }
`;

export const Descriptoin = styled.div`
  font-size: 14px;
  font-family: "Roboto";
  font-weight: 400;
  padding: 10px;
`;

export const ValueGroup = styled.div`
  border-top: 1px solid #d9d9d9;
  padding: 14px 10px;
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const ValueElm = styled.div`
  font-size: 14px;
  font-family: "Roboto";
  font-weight: 600;
`;

export const DividerH = styled(Divider).attrs(({ theme }) => ({
  height: "20px",
  width: "2px",
  color: "#d9d9d9",
}))``;

export const Overlay = styled.div`
  position: absolute;
  bottom: 0;
  background-color: rgba(255, 255, 255, 0.8);
  width: 100%;
  padding: 10px;
`;

export const OverlayContent = styled.div``;

export const Price = styled.div`
  color: #000000;
  font-size: 18px;
  font-family: "Roboto";
  font-weight: 600;
`;
export const PricePerSqm = styled.div`
  color: #000000;
  font-size: 14px;
  font-family: "Roboto";
  font-weight: 400;
`;
