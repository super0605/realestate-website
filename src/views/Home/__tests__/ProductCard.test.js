import React from "react";
import { mount, shallow } from "enzyme";
import ProductCard from "../ProductCard";
import NumberFormat from "react-number-format";

describe("ProductCard", () => {
  let props;
  let mountedProductCard;
  let wrapper;

  const renderWrapper = (bShallow) => {
    if (!mountedProductCard) {
      mountedProductCard = bShallow
        ? shallow(<ProductCard {...props} />)
        : mount(<ProductCard {...props} />);
    }
    return mountedProductCard;
  };

  beforeEach(() => {
    props = {
      title: "Lillian West",
      price: 30000,
      pricePerSqm: 4000,
      sqm: 120,
      numberOfBedrooms: 3,
      numberOfBathrooms: 2,
      imgUrl: "house.jpg",
    };

    wrapper = renderWrapper(true);
  });

  it("should render without problems", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render property title", () => {
    expect(wrapper.text().includes("Lillian West")).toBe(true);
  });

  it("should render property price and pricePerSqm", () => {
    expect(wrapper.find(NumberFormat).length).toBe(2);
  });

  it("should render property sqm", () => {
    expect(wrapper.text().includes("120 m²")).toBe(true);
  });

  it("should render property number of bedrooms", () => {
    expect(wrapper.text().includes("3 habs.")).toBe(true);
  });

  it("should render property number of bathrooms", () => {
    expect(wrapper.text().includes("2 baños")).toBe(true);
  });
});
