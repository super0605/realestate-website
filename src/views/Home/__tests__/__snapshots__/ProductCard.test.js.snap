// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard should render without problems 1`] = `
<styled.div>
  <styled.div>
    <img
      alt="example"
      src="house.jpg"
    />
    <styled.div>
      <styled.div>
        <styled.div>
          <NumberFormat
            allowEmptyFormatting={false}
            allowLeadingZeros={false}
            allowNegative={true}
            decimalScale={0}
            decimalSeparator="."
            displayType="text"
            fixedDecimalScale={false}
            isAllowed={[Function]}
            isNumericString={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseUp={[Function]}
            onValueChange={[Function]}
            prefix=""
            suffix="€"
            thousandSeparator={true}
            thousandsGroupStyle="thousand"
            type="text"
            value={30000}
          />
        </styled.div>
        <styled.div>
          <NumberFormat
            allowEmptyFormatting={false}
            allowLeadingZeros={false}
            allowNegative={true}
            decimalScale={3}
            decimalSeparator="."
            displayType="text"
            fixedDecimalScale={false}
            isAllowed={[Function]}
            isNumericString={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseUp={[Function]}
            onValueChange={[Function]}
            prefix=""
            suffix="€/m²"
            thousandSeparator={true}
            thousandsGroupStyle="thousand"
            type="text"
            value={4000}
          />
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    Lillian West
  </styled.div>
  <styled.div>
    <styled.div>
      <i
        className="fa fa-check-circle"
      />
       
      120
       m²
    </styled.div>
    <Styled(Divider) />
    <styled.div>
      <i
        className="fa fa-bed"
      />
       
      3
       habs.
    </styled.div>
    <Styled(Divider) />
    <styled.div>
      <i
        className="fa fa-bath"
      />
       
      2
       baños
    </styled.div>
  </styled.div>
</styled.div>
`;
